src/coprocessor/dag/expr/ctx.rs

```diff
@@ -99,6 +94,24 @@ impl EvalConfig {
         Ok(e)
     }
 
+    pub fn set_time_zone_by_name(&mut self, tz_name: &str) -> Result<()> {
+        let tz = Tz::from_tz_name(tz_name);
+        if tz.is_none() {
```

Comment:
You can you a match statement here and below to make it cleaner and avoid an unwrap.

```rust
match Tz::from_offset(tz_name) {
    Some(tz) => { self.tz = tz; Ok(()) },
    _ => Err(Error::invalid_timezone())
}
```

Reply:
good idea!